handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java

```diff
@@ -251,65 +262,71 @@ public String select(List<String> protocols) {
                 PREFERRED_APPLICATION_LEVEL_PROTOCOL);
-            ApplicationProtocolConfig serverApn = failingNegotiator(providerType.protocol(),
+            ApplicationProtocolConfig serverApn = failingNegotiator(param.providerType.protocol(),
                 APPLICATION_LEVEL_PROTOCOL_NOT_COMPATIBLE);
-            setupHandlers(serverApn, clientApn);
+            setupHandlers(param, serverApn, clientApn);
             assertTrue(serverLatch.await(2, TimeUnit.SECONDS));
             assertTrue(serverException instanceof SSLHandshakeException);
         } catch (SkipTestException e) {
             // ALPN availability is dependent on the java version. If ALPN is not available because of
             // java version incompatibility don't fail the test, but instead just skip the test
-            assumeNoException(e);
+            throw new AssumptionViolatedException("Not expected", e);
         }
     }
 
-    @Test
-    public void testAlpnCompatibleProtocolsDifferentClientOrder() throws Exception {
+    @MethodSource("newJdkParams")
+    @ParameterizedTest
+    public void testAlpnCompatibleProtocolsDifferentClientOrder(JdkSSLEngineTestParam param) throws Exception {
         try {
-            providerType.activate(this);
-            if (providerType == ProviderType.NPN_JETTY) {
+            param.providerType.activate(this);
+            if (param.providerType == ProviderType.NPN_JETTY) {
                 // This test only applies to ALPN.
-                throw tlsExtensionNotFound(providerType.protocol());
+                throw tlsExtensionNotFound(param.providerType.protocol());
             }
             // Even the preferred application protocol appears second in the client's list, it will be picked
             // because it's the first one on server's list.
             ApplicationProtocolConfig clientApn = acceptingNegotiator(Protocol.ALPN,
                 FALLBACK_APPLICATION_LEVEL_PROTOCOL, PREFERRED_APPLICATION_LEVEL_PROTOCOL);
             ApplicationProtocolConfig serverApn = failingNegotiator(Protocol.ALPN,
                 PREFERRED_APPLICATION_LEVEL_PROTOCOL, FALLBACK_APPLICATION_LEVEL_PROTOCOL);
-            setupHandlers(serverApn, clientApn);
+            setupHandlers(param, serverApn, clientApn);
             assertNull(serverException);
             runTest(PREFERRED_APPLICATION_LEVEL_PROTOCOL);
         } catch (SkipTestException e) {
             // ALPN availability is dependent on the java version. If ALPN is not available because of
             // java version incompatibility don't fail the test, but instead just skip the test
-            assumeNoException(e);
+            throw new AssumptionViolatedException("Not expected", e);
         }
     }
 
-    @Test
-    public void testEnablingAnAlreadyDisabledSslProtocol() throws Exception {
-        testEnablingAnAlreadyDisabledSslProtocol(new String[]{}, new String[]{ SslProtocols.TLS_v1_2 });
+    @MethodSource("newTestParams")
```

Comment:
How come this one is `newTestParams` and not `newJdkParams`?

Reply:
See https://github.com/netty/netty/pull/11483#discussion_r668989254